wger/core/models.py

```diff
@@ -218,6 +221,31 @@ class UserProfile(models.Model):
     Default duration of workout pauses in the gym view
     """
 
+    @property
+    def is_trustworthy(self):
+        """
+        Returns true for trustworthy if they've had an account for more than three weeks.
+        More trustworthiness criteria could be added in future.
+        """
+
+        days_since_joined = datetime.date.today() - self.user.date_joined.date()
+        days_in_three_weeks = 21
+
+        if days_since_joined.days > days_in_three_weeks:
+            # perms will be updated only once, after they hit three week mark
+            image_perm = Permission.objects.get(codename='add_exerciseimage')
+            self.user.user_permissions.add(image_perm)
+
+        return days_since_joined.days > days_in_three_weeks
+
+    @property
+    def has_exercise_permission(self):
+        """Returns true if user has all the exercise permissions and not just one."""
+        if self.user.groups.filter(name='admin').exists() or self.user.groups.filter(
```

Comment:
I'm not sure if checking for the group is a good idea. While we don't do now, if we add in the future a new group that should also be able to edit exercises this would not work. Better to check for individual permissions

Reply:
Makes sense! I believe if the strategy gets changed to doing the permission check dynamically as you mentioned below this will be irrelevant anyway.